bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/api/ApiHandler.java

```diff
@@ -86,6 +88,21 @@ public ApiHandler(HttpClient httpClient, TimeZoneProvider timeZoneProvider) {
                 .registerTypeAdapter(ForegroundApp.class, new ForegroundAppDeserializer())
                 .registerTypeAdapter(List.class, new ListDeserializer()).serializeNulls()
                 .registerTypeAdapterFactory(new StrictEnumTypeAdapterFactory()).create();
+        httpClient = httpClientFactory.createHttpClient(FreeboxOsBindingConstants.BINDING_ID);
+        httpClient.setResponseBufferSize(RESPONSE_BUFFER_SIZE);
+        try {
+            httpClient.start();
+        } catch (Exception e) {
+            throw new IllegalArgumentException("Unable to start httpClient", e);
+        }
+    }
+
+    public void dispose() {
+        try {
+            httpClient.stop();
+        } catch (Exception e) {
+            throw new IllegalArgumentException("Unable to stop httpClient", e);
```

Comment:
A simple DEBUG log would be sufficient.

Reply:
I guess this is more than unlikely to happen but when it happens I suppose it is suitable to have something more visible than a DEBUG log, no ?

Reply:
When looking at other bindings, I see both **warn** and **debug** logging used as well as fully ignoring it. So logging a warning wouldn't be as controversial as throwing an `IllegalArgumentException`. 😉